pageserver/src/tenant/timeline/offload.rs

```diff
@@ -33,6 +34,21 @@ pub(crate) async fn offload_timeline(
     debug_assert_current_span_has_tenant_and_timeline_id();
     tracing::info!("offloading archived timeline");
 
+    if !timeline.remote_client.no_pending_work() {
+        return Err(OffloadError::RemoteStorage(anyhow!(
+            "cannot offload timeline with pending uploads"
```

Comment:
I think this should be a separate `OffloadError` variant, that we silently ignore in compaction (or log an info line).